dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2639,327 @@ void FileSprayer::pushParts()
     performTransfer();
 }
 
+using namespace Azure::Storage;
+using namespace Azure::Storage::Files;
+using namespace Azure::Storage::Blobs;
+
+enum ApiCopyStatus { Pending, Success, Failed, Aborted };
+interface IAPICopyClient : implements IInterface
+{
+    virtual ApiCopyStatus startCopy(const char *target, const char * source) = 0;
+    virtual ApiCopyStatus getProgress(CDateTime & dateTime, int64_t & outputLength) = 0;
+    virtual ApiCopyStatus abortCopy() = 0;
+    virtual ApiCopyStatus getStatus() const = 0;
+};
+
+class CApiCopyClient : implements IAPICopyClient, public CInterface
+{
+protected:
+    ApiCopyStatus status;
+public:
+    IMPLEMENT_IINTERFACE
+    virtual ApiCopyStatus getStatus() const
+    {
+        return status;
+    }
+    static bool buildAzureStorageURIBase(IStoragePlane * storagePlane, StringBuffer & uri)
+    {
+        const char *accountName = storagePlane->queryStorageApiAccount();
+        if (isEmptyString(accountName))
+        {
+            UWARNLOG("Unable to API copy: no account name provided for storage plane");
```

Comment:
should useAPICopy() have already have verified that it does have an a/c name by this stage? And if so, this should prob. be an exception.

Reply:
I've moved the checks to useApiCopy() and removed the test here.  I don't think it's necessary to assertex(!isEmptyString(accountName)) here.